pkg/cmd/ban.go

```diff
@@ -11,26 +14,170 @@ import (
 	"github.com/kubesaw/ksctl/pkg/ioutils"
 
 	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
 )
 
+const (
+	menuKey       string = "menu.json"
+	configMapName string = "ban-reason-config"
+)
+
+// Menu contains all the fields present in the source JSON file needed to load up options in the interactive menu
+type Menu struct {
+	Kind        string   `json:"kind"`
+	Description string   `json:"description"`
+	Options     []string `json:"options"`
+}
+
+// getValuesFromConfigMap retrieves the configMap contents to build the interactive menus
+func getValuesFromConfigMap(ctx *clicontext.CommandContext) ([]Menu, error) {
+	cfg, err := configuration.LoadClusterConfig(ctx, configuration.HostName)
+	if err != nil {
+		return nil, err
+	}
+	cl, err := ctx.NewClient(cfg.Token, cfg.ServerAPI)
+	if err != nil {
+		return nil, err
+	}
+
+	configMap := &corev1.ConfigMap{}
+	err = cl.Get(context.TODO(), types.NamespacedName{
+		Name:      configMapName,
+		Namespace: cfg.OperatorNamespace,
+	}, configMap)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var menus []Menu
+	if menuJSON, exists := configMap.Data[menuKey]; exists && menuJSON != "" {
+		//var menus []Menu
+		if err := json.Unmarshal([]byte(menuJSON), &menus); err != nil {
+			return nil, fmt.Errorf("the %s key in the configmap doesn't contain a valid JSON format to render menus: %w", menuKey, err)
+		}
+	}
+
+	return menus, nil
+}
+
+// BanMenu displays an interactive menu for selecting banning reasons
+func BanMenu(ctx *clicontext.CommandContext, runForm runFormFunc, cfgMapContent []Menu) (map[string]string, error) {
+
+	// Map to store user's answers
+	answers := make(map[string]string)
+
+	if len(cfgMapContent) > 0 {
+		for _, item := range cfgMapContent {
+			var choice string
+			options := make([]huh.Option[string], len(item.Options))
+			for i, opt := range item.Options {
+				options[i] = huh.Option[string]{Key: opt, Value: opt}
+			}
+
+			form := huh.NewForm(
+				huh.NewGroup(
+					huh.NewSelect[string]().
+						Title(item.Description).
+						Options(options...).
+						Value(&choice),
+				),
+			)
+
+			if err := runForm(form); err != nil {
+				return nil, err //fmt.Errorf("failed to show interactive menu: %w", err)
```

Comment:
let's clean up all these zombies :ghost: 
```suggestion
				return nil, err
```